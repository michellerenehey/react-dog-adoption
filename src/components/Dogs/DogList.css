.dog-card {
  border: 1px black solid;
  border-radius: 5px;
  margin: 5px;
  width: 300px;
  height: 300px;
}
.dog-list {
  text-decoration: none;
  color: black;
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
}

.dog-list > img {
  height: 150px;
  border-radius: 50%;
}

.dog-card-p {
  text-align: center;
}
