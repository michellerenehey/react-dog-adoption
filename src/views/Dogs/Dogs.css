.dogs-style {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
