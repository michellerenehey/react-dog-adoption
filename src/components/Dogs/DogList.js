import { Link } from 'react-router-dom';
import './DogList.css';

export default function DogsList({ id, name, breed, image }) {
  return (
    <div className="dog-card">
      <Link to={`/dogs/${id}`} className="dog-list">
        <h3>Meet {name}</h3>
        <img src={image} />
        <p className="dog-card-p">
          {name} is a(n) {breed}
        </p>
      </Link>
    </div>
  );
}
