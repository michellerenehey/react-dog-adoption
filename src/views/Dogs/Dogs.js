import { fetchDogs } from '../../services/dogs';
import { useEffect, useState } from 'react';

import DogList from '../../components/Dogs/DogList';
import './Dogs.css';

export default function Dogs() {
  const [dogs, setDogs] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
      setLoading(false);
    };
    fetchData();
  }, []);

  return (
    <div>
      {loading ? (
        <p>...page is loading</p>
      ) : (
        <div className="dogs-style">
          {dogs.map((dog) => (
            <DogList key={dog.id} {...dog} />
          ))}
        </div>
      )}
    </div>
  );
}
