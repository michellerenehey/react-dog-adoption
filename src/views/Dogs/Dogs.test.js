import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Dogs from './Dogs';

test('should render list of dogs', async () => {
  const { container } = render(
    <MemoryRouter>
      <Dogs />
    </MemoryRouter>
  );
  await screen.findByText('Meet Franny');
  expect(container).toMatchSnapshot();
});
