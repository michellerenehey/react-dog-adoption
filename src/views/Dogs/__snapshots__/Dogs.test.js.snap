// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of dogs 1`] = `
<div>
  <div>
    <div
      class="dogs-style"
    >
      <div
        class="dog-card"
      >
        <a
          class="dog-list"
          href="/dogs/6"
        >
          <h3>
            Meet 
            Etta
          </h3>
          <img
            src="https://placedog.net/1000/563?id=21"
          />
          <p
            class="dog-card-p"
          >
            Etta
             is a(n) 
            Frenchie
          </p>
        </a>
      </div>
      <div
        class="dog-card"
      >
        <a
          class="dog-list"
          href="/dogs/4"
        >
          <h3>
            Meet 
            Kaya
          </h3>
          <img
            src="https://placedog.net/1000/563?id=118"
          />
          <p
            class="dog-card-p"
          >
            Kaya
             is a(n) 
            German shepherd mix
          </p>
        </a>
      </div>
      <div
        class="dog-card"
      >
        <a
          class="dog-list"
          href="/dogs/18"
        >
          <h3>
            Meet 
            Not a Dog
          </h3>
          <img
            src="https://placekitten.com/200/300"
          />
          <p
            class="dog-card-p"
          >
            Not a Dog
             is a(n) 
            Cat
          </p>
        </a>
      </div>
      <div
        class="dog-card"
      >
        <a
          class="dog-list"
          href="/dogs/3"
        >
          <h3>
            Meet 
            Poppy
          </h3>
          <img
            src="https://placedog.net/1000/563?id=125"
          />
          <p
            class="dog-card-p"
          >
            Poppy
             is a(n) 
            Border collie mix
          </p>
        </a>
      </div>
      <div
        class="dog-card"
      >
        <a
          class="dog-list"
          href="/dogs/1"
        >
          <h3>
            Meet 
            Sitka
          </h3>
          <img
            src="https://placedog.net/1000/563?id=73"
          />
          <p
            class="dog-card-p"
          >
            Sitka
             is a(n) 
            Cattle Dog Mix
          </p>
        </a>
      </div>
      <div
        class="dog-card"
      >
        <a
          class="dog-list"
          href="/dogs/2"
        >
          <h3>
            Meet 
            Xavi
          </h3>
          <img
            src="https://placedog.net/1000/563?id=83"
          />
          <p
            class="dog-card-p"
          >
            Xavi
             is a(n) 
            Cattle dog mix
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
